includes/plugins/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -119,5 +130,481 @@ public static function my_account_menu_items( $items ) {
 		unset( $items['customer-logout'] );
 		return $items;
 	}
+
+	/**
+	 * Remove required fields from the account settings form.
+	 *
+	 * @param array $required_fields The required fields.
+	 * @return array The required fields.
+	 */
+	public static function account_settings_required_fields( $required_fields ) {
+		unset( $required_fields['account_display_name'] );
+		return $required_fields;
+	}
+
+	/**
+	 * Intercept a password reset request from Account Settings page.
+	 * The key must be generated before the user's password can be reset,
+	 * but not before submitting the form, otherwise it will break the Forgot Password flow.
+	 */
+	public static function maybe_generate_password_reset_key() {
+		// Only if the user is logged in and a reader.
+		$user = \wp_get_current_user();
+		if ( ! \is_user_logged_in() || ! Reader_Activation::is_user_reader( $user ) ) {
+			return;
+		}
+
+		// Only if updating password from Account Settings page.
+		$action = filter_input( INPUT_POST, 'action', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		$nonce  = filter_input( INPUT_POST, 'woocommerce-reset-password-nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		if ( self::RESET_PASSWORD_ACTION !== $action || empty( $nonce ) ) {
+			return;
+		}
+
+		// Generate a password reset key.
+		$key   = \get_password_reset_key( $user );
+		$login = $user->data->user_login;
+		if ( \is_wp_error( $key ) ) {
+			return;
+		}
+
+		// Pass the key and login to the posted data.
+		$_POST['reset_key']   = $key;
+		$_POST['reset_login'] = $login;
+	}
+
+	/**
+	 * Check if the user has a valid password reset key.
+	 *
+	 * @return bool True if the user has a valid password reset key, false otherwise.
+	 */
+	public static function check_password_reset_key() {
```

Comment:
I think this and the method below could benefit from the readability of guard clauses, like it's being done in other methods for this class.

Reply:
24392523 refactors all of the methods in the `_V1` classes to use the same guard clause patterns. In the specific case of the `My_Account_UI_V1_Passwords::check_password_reset_key()` method, I'm not sure we need the guard clauses as it's a read-only method and is only called by `My_Account_UI_V1_Passwords::redirect_reset_password_link()`, which does check all conditions.